src/middleware/healthRoutes/index.js

```diff
@@ -0,0 +1,54 @@
+import { getData } from '../../apiProvider';
+
+export const status = {
+  healthy: {
+    code: 200,
+    message: 'Healthy',
+  },
+  degraded: {
+    code: 200,
+    message: 'Degraded',
+  },
+  unhealthy: {
+    code: 503,
+    message: 'Unhealthy',
+  },
+};
+
+const getReadyStatus = async ({ dependencies, logger }) => {
+  const dependenciesPromises = await dependencies.map(async (dependency) => {
+    try {
+      await getData({ endpoint: dependency.endpoint, logger });
+      logger.info(`${dependency.name} is healthy`);
+      return 'healthy';
+    } catch (e) {
+      if (dependency.critical) {
+        logger.error(`critical dependency ${dependency.name} is not healthy`);
```

Comment:
should probably say `unhealthy` instead of `not healthy` incase someone wants to search the logs